src/test/java/com/fittracker/fittracker/service/WeightServiceTest.java

```diff
@@ -185,5 +193,60 @@ void givenNoDateFound_shouldThrowException() {
             verifyNoMoreInteractions(weightRepository);
         }
     }
+    @Nested
+    class FindByDateRange {
+        private final LocalDate endDate = TEST_DATE.plusDays(3);
+        private final Weight endWeight = new Weight(endDate,TEST_VALUE,TEST_UUID);
+        @Test
+        void givenStartDateAfterEndDate_shouldThrowInvalidDateRangeException() {
+            assertThatThrownBy(()->weightService.findByDateRange(LocalDate.of(2023,5,1),LocalDate.of(2023,1,1)))
+                    .isInstanceOf(InvalidDateRangeException.class)
+                    .hasMessageContaining("Start date cannot be after end date");
+
+            verifyNoInteractions(weightRepository);
+        }
+
+        @Test
+        void givenValidDateRange_shouldReturnListOfWeightResponses() {
+            when(weightRepository.findByDateBetweenAndUserId(TEST_DATE, endDate,TEST_UUID))
```

Comment:
let's use `any()` when setting up mocks

Reply:
Right, we are arleady verifying that call are made iwth correct arguments by using: `verify(objectName).methodName(args)`